### ISSUE TITLE
Add new function to validate fuel types and set output strings

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#7621]
**This is part 3 of 4: Part 1 = #7532, Part 2 = #7523 , and Part 4 is #7622.**

Issue overview
--------------
The goal of this issue to to remove all if blocks for fuel type in the input processing functions and replace them with a standard function call.

This issue should only standardize the fuel type choice input processing and introduce *no output changes* - that can wait for #?????.

### Background ###
#7523 standardized all fuel type inputs to a single list with no synonyms. Synonyms were removed from all (most?) input processing functions. #7523 introduces no output changes.

Some input processing using the functions `DataGlobalConstants::AssignResourceTypeNum` and `DataGlobalConstants::GetResourceTypeChar` to validate fuel types and set the standard strings for output variables and meter resource types. But many input processing functions still have their own if blocks to test for "NaturalGas" "Electricity" etc. This issue is to use a standard fuel type function everywhere as appropriate. 

### More Details ###
1. Given that fuel types are a subset of resource types, `AssignResourceTypeNum` could be modified to take an argument to only allow fuels (no other resource types) or it may be best to have a new function(s) just for fuel type inputs.

2. In general, objects with a fuel type choice need to validate the input choice and assign strings to use for output variables and the correct resource type to use for meters.

3. `AssignResourceTypeNum` and `GetResourceTypeChar` may be ripe for some refactoring.

4. For this first pass with no output diffs, the mappings are shown below. One might be tempted to say we don't need three strings. That's true for some fuel types, like Diesel, but FuelOilNo1 will ultimately map to "Fuel Oil No 1" for output variables and table headings in the last phase of this. For now some of the old names are kept to produce no diffs.

**Input choice, Meter Resource, Output Variable**
Electricity, Electricity, Electric
NaturalGas, Gas, Gas
Propane, Propane, Propane
Diesel, Diesel, Diesel
Gasoline, Gasoline, Gasoline
FuelOilNo1, FuelOil#1, FuelOil#1
FuelOilNo2, FuelOil#2, FuelOil#2
Coal, Coal, Coal
OtherFuel1, OtherFuel1, OtherFuel1
OtherFuel2, OtherFuel2, OtherFuel2
Water, Water, Water

4. For every type except NaturalGas, the meter resource type is identical to the input fuel type. And ultimately, NaturalGas will be the same, also. So perhaps the input fuel type can simply be passed directly to the `ResourceTypeKey` in `SetupOutputVariable`.

5. Other related functions are "GetStandardMeterResourceType" which is called from during SetupOutputVariable processing. So, this function could stay as-is and take care of the one oddball meter resource "Gas" which ultimately will move to "NaturalGas". This function should be evaluated for refactoring as well. Even though the input choices should be fixed in the idd/schema, this function should probably remain to trap any oddball nonstandard fuel types that might get added. And it can clean up any upper/lower case inconsistencies.

6. There may be some subtleties here that gum up the works for the no-diff case. For example electricity use is typically "Electric Power" and "Electric Energy" in output variables. While most (all?) other fuels are "Gas Rate", "Gas Energy".

Lots of scattered info here - sorry.

### Details
Some additional details for this issue (if relevant):
 - Followup for #5941.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by dareumnam** [comment]
I am wondering if someone already started working on this issue. If not, I would like to take a look at this.

**Comment by mjwitte** [comment]
@dareumnam I have not done anything on this yet.

**Comment by dareumnam** [comment]
@mjwitte Thanks for letting me know. I'll take a look at this.

**Comment by dareumnam** [comment]
As Michael said in the last iteration call, fuel-types in each component were input by key-type choices, so users barely input wrong fuel types. So one of the issues about standardizing a function for fuel types is when users leave a fuel type input black.
 - Components can be classified by how they treat the blank fuel type input in input processing.
1. Components that have code lines designating a specific default fuel type for a blank input. (Default fuels depend on the components)
Example) CTElectircGenerator.cc
https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/CTElectricGenerator.cc#L312-L346
2. Components that have code lines showing severe error messages for invalid input (including blank)
Example) ChillerGasAbsorption.cc
https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ChillerGasAbsorption.cc#L455-L487
3. Components that have code lines showing error messages first and designating a default fuel type for a blank input
Example)
Boilers.cc https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/Boilers.cc#L238-L289
 - Regardless of this, in the IDD file, some components designated a default fuel type choice and the others do not.

It would be very helpful if you let me know your insight about this - unifying the way each component treats the blank input & add one standardized function, **or** separating these cases & add two or three standardized functions.  @mjwitte @Myoldmopar @mitchute

**Comment by mjwitte** [comment]
My preference would be that blank is never sent to the new function.
1. If the fuel type field is tagged \required-field in the IDD, then the input processor will reject a blank with a fatal error.
2. If the fuel type field has a \default in the IDD, then the input processor should fill in the default, so again, blank should not make it to the new function. However, this requires that \min-fields for the object includes the fuel type field. If this is not the case, we'll need to review those objects individually.
3. If the fuel type field does not have a default and is not required, then we should add one or the other to the IDD.

So, basically, my suggestion is that we review the IDD and make any necessary changes such that a blank fuel type never makes it into the field value in the getinput functions.
The function, should have a fatal error directed at developers (a user should never encounter it) that trips if the incoming fuel type does not match one of the standard types. Blank would also be fatal.
